### PR TITLE
acme_challenge_cert_helper: add support for IP identifiers

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Adds support for IP identifiers to acme_challenge_cert_helper. Uses the extended TLS-ALPN-01 challenge data returned by #53660.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
acme_challenge_cert_helper

### CODE REVIEW

File: lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py [hunk 1]
```diff
@@ -206,11 +222,16 @@ def main():
 
         # Some common attributes
         domain = to_text(challenge_data['resource'])
-        subject = issuer = cryptography.x509.Name([
-            cryptography.x509.NameAttribute(cryptography.x509.oid.NameOID.COMMON_NAME, domain),
-        ])
+        type, identifier = to_text(challenge_data.get('resource_original', 'dns:' + challenge_data['resource'])).split(':', 1)
```
Comment (resmo) [inline lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py]: not a blocker, but I would rename `type` to `_type`, type is a reserved word in python

Comment (felixfontein) [reply lib/ansible/modules/crypto/acme/acme_challenge_cert_helper.py]: Good point! I'll fix that in this PR and #53660.

**Review by resmo — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
@resmo thanks for reviewing and merging!

**Pull request merged**